awx/api/serializers.py

```diff
@@ -2176,11 +2176,17 @@ class Meta:
 class InventoryUpdateSerializer(UnifiedJobSerializer, InventorySourceOptionsSerializer):
 
     custom_virtualenv = serializers.ReadOnlyField()
+    source_project = serializers.SerializerMethodField(
+        help_text=_('The project used for this job.'),
+    )
 
     class Meta:
         model = InventoryUpdate
-        fields = ('*', 'inventory', 'inventory_source', 'license_error', 'source_project_update',
-                  'custom_virtualenv', '-controller_node',)
+        fields = ('*', 'inventory', 'inventory_source', 'license_error', 'source_project',
+                  'source_project_update', 'custom_virtualenv', '-controller_node',)
+
+    def get_source_project(self, obj):
+        return getattrd(obj, 'source_project_update.unified_job_template.id', None)
```

Comment:
This does incur additional queries, which is why I hesitated from doing it myself. You can add a new ForeignKey field, but that also automatically adds an index... which increases the memory footprint. My concern for these serializers is never the detail view, or a listing of project updates. My concern is always the `/api/v2/unified_jobs/` endpoint, which the UI hits aggressively. The additional queries will be hit for each SCM inventory update anywhere in the list as a user looks at the JOBS page.

Reply:
I see. How about we only do this for details views, then? I think there's a `is_detail` property we can leverage.

Reply:
Yes, that is one workable option.

Reply:
alright - done. These are now only on the details serializer. I did some retesting and everything still looks good.